lifecycle_test.go

```diff
@@ -456,6 +456,83 @@ func TestMergePortBindings(t *testing.T) {
 	}
 }
 
+func TestPortMappingCheck(t *testing.T) {
+	makePortMap := func(ports []string) nat.PortMap {
+		out := make(nat.PortMap)
+		for _, port := range ports {
+			// We don't care about the actual binding in this test
+			out[nat.Port(port)] = make([]nat.PortBinding, 0)
+		}
+		return out
+	}
+
+	tests := []struct {
+		name                  string
+		exposedAndMappedPorts nat.PortMap
+		exposedPorts          []string
+		expectError           bool
+	}{
+		{
+			name:                  "No protocol given",
```

Comment:
nit: using test names with spaces makes it hard to find when doing a search due to space to underscore replacement.

I tend to use `-` instead and keep succinct e.g.
```suggestion
			name:                  "no-protocol",
```

Reply:
https://github.com/testcontainers/testcontainers-go/pull/2658/commits/5fe2f77b1c6101b2c550f182ec4021d828ff5bd0

Reply:
I do not have a strong opinion about using white spaces, as it makes it easier to "write" test names. OTOH, if that makes "reading/searching" test names easier, I'm fine with it